python/semantic_kernel/connectors/memory/postgres/postgres_memory_store.py

```diff
@@ -0,0 +1,524 @@
+        default_dimensionality: int,
+        min_pool: int,
+        max_pool: int,
+        schema: str = DEFAULT_SCHEMA,
+        logger: Optional[Logger] = None,
+    ) -> None:
+        """Initializes a new instance of the PostgresMemoryStore class.
+
+        Arguments:
+            connection_string {str} -- The connection string to the Postgres database.
+            default_dimensionality {int} -- The default dimensionality of the embeddings.
+            min_pool {int} -- The minimum number of connections in the connection pool.
+            max_pool {int} -- The maximum number of connections in the connection pool.
+            schema {str} -- The schema to use. (default: {"public"})
+            logger {Optional[Logger]} -- The logger to use. (default: {None})
+        """
+        if default_dimensionality > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {default_dimensionality} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if default_dimensionality <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
+
+        self._connection_string = connection_string
+        self._default_dimensionality = default_dimensionality
+        self._connection_pool = ConnectionPool(
+            self._connection_string, min_size=min_pool, max_size=max_pool
+        )
+        self._schema = schema
+        atexit.register(self._connection_pool.close)
+        self._logger = logger or NullLogger()
+
+    async def create_collection_async(
+        self,
+        collection_name: str,
+        dimension_num: Optional[int] = None,
+    ) -> None:
+        """Creates a new collection.
+
+        Arguments:
+            collection_name {str} -- The name of the collection to create.
+            dimension_num {Optional[int]} -- The dimensionality of the embeddings. (default: {None})
+            Uses the default dimensionality when not provided
+
+        Returns:
+            None
+        """
+        if dimension_num is None:
+            dimension_num = self._default_dimensionality
+        if dimension_num > MAX_DIMENSIONALITY:
+            raise ValueError(
+                f"Dimensionality of {dimension_num} exceeds "
+                + f"the maximum allowed value of {MAX_DIMENSIONALITY}."
+            )
+        if dimension_num <= 0:
+            raise ValueError("Dimensionality must be a positive integer. ")
+
+        with self._connection_pool.connection() as conn:
+            conn.execute("CREATE EXTENSION IF NOT EXISTS vector")
+            with conn.cursor() as cur:
+                cur.execute(
+                    SQL(
+                        """
+                        CREATE TABLE IF NOT EXISTS {scm}.{tbl} (
+                            key TEXT PRIMARY KEY,
+                            embedding vector({dim}),
+                            metadata JSONB,
+                            timestamp TIMESTAMP
```

Comment:
C# uses BIGINT type to unix to store timestamp, I am considering whether to use TIMESTAMP to store it. The postgres timestamp requires that it be stored in UTC time, and when you query it again, it returns the utc time, with no way of knowing the actual time zone when it was stored. How do you think about it?

Reply:
This was something that I saw when I was looking at the most recent version of the C# implementation, so I am really glad that you brought this up! You are correct based on the docs, Postgres has the concept of both timestamp without time zone and timestamp with time zone, and this does not affect the storage as both are ultimately stored as 8-byte integers in UTC. 

[Time Stamps](https://www.postgresql.org/docs/current/datatype-datetime.html#:~:text=8.5.1.3.%C2%A0Time%20Stamps)

This would mean that we would need to be contextually aware of the timezone of the client (which sounds like we are in the C# implementation) and sprinkle in the `timezone(zone, timestamp)` or 'AT TIME ZONE' conversion functions to push the conversion work onto the db.

[Time Stamp Functions](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT)

Ultimately, I think that we should go with `timestamp with time zone`and use the built-in functions where appropriate. Do you agree?

Reply:
Initial stab at handling this is in the conde now.

Reply:
I have changed the `MemoryRecord `implementation to use `datetime` instead of `str` for the `timestamp`. With this in place I have changed the `PostresqlMemoryStore` and tests to account for this. 

Additionally, I have made the necessary changes to the `ChromaMemoryStore` and `PineconeMemoryStore` and tests to work with this new structure but am holding off on pushing this since it is outside of the scope of this PR. I am happy to include it if requested. 

The only memory store that I havent created an update for is the `WeaviateMemoryStore`, since I dont have it installed at this time.

Reply:
As a best practice, I think datetimes in database should be stored in UTC and converted on client side to appropriate timezone. In other words, it's a client responsibility to handle datetime conversion.

Semantic Kernel, as SDK, shouldn't be aware of timezones at all. That means, if customer uses UTC to store datetimes in database, then SK puts a record with timestamp in UTC. If customer decided to use local timezone to store information - then SK puts a record with timestamp in local timezone. In other words, we should pass timezone in the same format as we received from SDK user and we should not perform any conversions.

As an example, in case there will be some default logic, when we put some default timestamp (for example `DateTime.UtcNow`) - the timezone should always be in UTC. Although even in this case I would probably set default timestamp value as `null`, but it depends on scenario.

@cschadewitz I see some datetime conversion in `postgres_memory_store.py`, so I think we need to remove it, since it's SDK consumers responsibility. Please let me know if that makes sense.

We will probably need to make some changes in C# kernel to make timestamp field as timezone-agnostic, but it's definitely a scope of separate PR.

Reply:
That is fine with me if we are passing the responsibility off to the consumer of the SDK.